spring-ai-core/src/main/java/org/springframework/ai/document/Document.java

```diff
@@ -149,6 +189,15 @@ public Map<String, Object> getMetadata() {
 		return this.metadata;
 	}
 
+	@Nullable
+	public Double getScore() {
+		return this.score;
+	}
+
+	public void setScore(@Nullable Double score) {
+		this.score = score;
+	}
```

Comment:
Should the document object be immutable?  If we want to update a Document with a score we should use a builder that takes the existing document and then call the 'score' builder method?

Reply:
The post-retrieval steps in a RAG flow would all modify the Documents in some way, including the score, the content and the metadata. I guess we could create new instances on each hop to the next step in the flow. But we'd need several builders based on the type of field we change. Should I go with that?